pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -231,28 +213,54 @@ type ReconcileIntrusionDetection struct {
 	licenseAPIReady *utils.ReadyFlag
 	dpiAPIReady     *utils.ReadyFlag
 	tierWatchReady  *utils.ReadyFlag
-	usePSP          bool
+	multiTenant     bool
 	elasticExternal bool
 }
 
+func GetIntrusionDetection(ctx context.Context, cli client.Client, mt bool, ns string) (*operatorv1.IntrusionDetection, error) {
```

Comment:
Is this used outside of this file? If it is, it might be better in the utils or some other library package (and if not it probably doesn't need to be exported.

Reply:
The pattern we have established for other things (I think) is that these live within the controller that owns that CR. Pretty sure it doesn't need to be exported though